internal/agent/hooks/kcrypt_uki.go

```diff
@@ -143,5 +149,32 @@ func (k KcryptUKI) Run(c config.Config, _ v1.Spec) error {
 	if err != nil {
 		return err
 	}
+
+	// Copy logs to persistent partition
```

Comment:
this makes more sense to me, we just open them once as we need to copy back the oem config and we take advantage of that time slot to copy the logs over if possible.
This reduces the lines of code needed for this, the lines of code that play with mounting unmounting and unlocking and the time the partitions are unlocked